src/codeu/chat/util/MorseConverter.java

```diff
@@ -0,0 +1,217 @@
+
+/**
+ * Morse code converter. Note that it only works with the English language and
+ * can only translate the 26 letter alphabet and numbers 0-9. Note that Morse
+ * code works by representing each char with a combination of "." and/or "–".
+ * Words are separated by "/". There is no ending punctuation in Morse code.
+ *
+ * @author Lauren
+ *
+ */
+public final class MorseConverter {
+  private static final Map<Character, String> CODE = ImmutableMap
+      .<Character, String>builder().put('a', ".–").put('b', "–...")
+      .put('c', "–.–.").put('d', "–..").put('e', ".").put('f', "..–.")
+      .put('g', "––.").put('h', "....").put('i', "..").put('j', ".–––")
+      .put('k', "–.–").put('l', ".–..").put('m', "––").put('n', "–.")
+      .put('o', "–––").put('p', ".––.").put('q', "––.–").put('r', ".–.")
+      .put('s', "...").put('t', "–").put('u', "..–").put('v', "...–")
+      .put('w', ".––").put('x', "–..–").put('y', "–.––").put('z', "––..")
+      .put('0', "–––––").put('1', ".––––").put('2', "..–––")
+      .put('3', "...––").put('4', "....–").put('5', ".....")
+      .put('6', "–....").put('7', "––...").put('8', "–––..")
+      .put('9', "––––.").build();
+
+  private MorseConverter() {
+  }
+
+  /**
+   * Convert a word into morse code. Only works for english alphabet.
+   *
+   * @param word
+   *          String
+   * @return String in morse
+   */
+  public static String wordToMorse(String word) {
+    char[] characters = word.toLowerCase(Locale.ENGLISH).toCharArray();
+    StringBuilder strBuild = new StringBuilder();
+    for (int i = 0; i < characters.length; i++) {
+      strBuild.append(CODE.get(characters[i]));
+      strBuild.append(" ");
+    }
+    strBuild.append("/");
+    return strBuild.toString();
+  }
+
+  /**
+   * Converts a paragraph into morse code. Delineates all of the punctuation.
+   *
+   * @param paragraph
+   *          String
+   * @return String of morse
+   */
+  public static String paragraphToMorse(String paragraph) {
+    String[] words = paragraph.split("\\W+");
+    StringBuilder strBuild = new StringBuilder();
+    for (String word : words) {
+      strBuild.append(wordToMorse(word));
+    }
+    return strBuild.toString();
+  }
+
+  /**
+   * Return the letter associated with the given morse.
+   *
+   * @param morse
+   *          String
+   * @return String of english letter
+   */
+  public static char morseToLetter(String morse) {
+    if (!morseCheck(morse)) {
+      throw new IllegalArgumentException(
+          "ERROR: string of morse must be only . and –");
+    }
+    Iterator<Entry<Character, String>> itr = CODE.entrySet().iterator();
+    while (itr.hasNext()) {
+      Entry<Character, String> entry = itr.next();
+      if (entry.getValue().equals(morse)) {
+        return entry.getKey();
+      }
+    }
+    return '\0';
+  }
+
+  private static boolean morseCheck(String morse) {
+    char[] check = morse.toCharArray();
+    for (int i = 0; i < check.length; i++) {
```

Comment:
to iterate over each char, you could write it more compactly as `for (char c : morse.toCharArray())`